src/util/nft.js

```diff
@@ -283,6 +285,59 @@ export const getNFTClassesRespectDualPrefix = async (axios, owner) => {
   return arraysOfNFTClasses.flat();
 };
 
+export function getEventKey(event) {
+  const { classId, nftId, txHash } = event;
+  return `${classId}-${nftId}-${txHash}`;
+}
+
+async function getPurchasePrice(axios, nftId, classId, txHash) {
+  try {
+    const { data } = await axios.get(api.getChainRawTx(txHash));
+    const { price } = data.tx.body.messages.find(
+      m =>
+        m['@type'] === '/likechain.likenft.v1.MsgBuyNFT' &&
+        m.class_id === classId &&
+        m.nft_id === nftId
+    );
+    return new BigNumber(price).shiftedBy(-9).toNumber();
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(error);
+    return null;
+  }
+}
+
+export async function getPurchasePriceMap(axios, history) {
+  const priceMap = new Map();
+  await Promise.all(
+    history.map(async e => {
+      const { event, txHash, classId, nftId } = e;
+      if (event === 'buy_nft') {
+        const price = await getPurchasePrice(axios, nftId, classId, txHash);
+        const key = getEventKey(e);
+        priceMap.set(key, price);
+      }
+    })
+  );
+  return priceMap;
```

Comment:
I don't think `Array.prototype.map` should have side effects (i.e. setting `priceMap` in this case)

```suggestion
  return new Map(
    await Promise.all(
      history.filter(e => e.event === 'buy_nft').map(async e => {
        const { txHash, classId, nftId } = e;
        const price = await getPurchasePrice(axios, nftId, classId, txHash);
        const key = getEventKey(e);
        return [key, price];
      })
    )
  );
```